File: Stock/src/main/java/net/lahlalia/stock/entities/Bac.java
package net.lahlalia.stock.entities;

import jakarta.persistence.*;
import lombok.*;
import net.lahlalia.stock.dtos.Product;
import org.springframework.format.annotation.DateTimeFormat;

import java.util.Date;
import java.util.List;

@Entity
@Getter
@Setter
@ToString
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Bac {
    @Id
    private String idBac;
    private double capacity;
    private double totalImpom;
    private boolean status;
    @DateTimeFormat(pattern = "dd-MM-yyyy")
    private Date dateOuverture;
    private double capacityUsed;
    private Long idProduct;

    @OneToMany(mappedBy = "bac")
    private List<EntreSortie> entre_sortie;

    @ManyToOne
    private Depot depot;



}

File: vivo-frontend/src/app/models/Bac.ts
import {Regime} from "./Regime";
import {Product} from "./Product";

export class Bac{
  idBac: String|undefined;
  capacity: number|undefined;
  totalImpom: number|undefined;
  status: boolean|undefined;
  dateOuverture: Date|undefined;
  capacityUsed: number|undefined;
  idProduct: number|undefined;
  product?: Product;
  creux: number|undefined;


}

